pkg/service/grantsvc/granter.go

```diff
@@ -137,6 +137,47 @@ func (g *Granter) RevokeGrant(ctx context.Context, opts RevokeGrantOpts) (*acces
 	return nil, errors.New("unhandled response code")
 }
 
+// validate grant runs all the checks that will need to occur when creating a real grant to validate its success
+func (g *Granter) ValidateGrant(ctx context.Context, opts CreateGrantOpts) error {
+	q := &storage.GetUser{
```

Comment:
I propose that you change the input to this method to avoid needing to call dynamo db again to fetch the user as its already called in the create grant method.

I think you can also make this a private method as it will only be used by createGrant.

Because the types of PostGrantsJSONRequestBody and ValidateRequestToProviderJSONRequestBody are identical, could you use a shared type in the open API schema instead?

That way you can reduce the code duplication, and potentially remove this validation function all together and just have the call to validate as part of create grant